status/reporters.go

```diff
@@ -155,6 +330,207 @@ func (r *GitHubReporter) getToken(ctx context.Context, object client.Object, nam
 	return string(token), nil
 }
 
+func (csu *CommitStatusUpdater) getAllCommitStatuses(ctx context.Context) ([]*ghapi.RepoStatus, error) {
+	if len(csu.allCommitStatusesCache) == 0 {
+		allCommitStatuses, err := csu.ghClient.GetAllCommitStatusesForRef(ctx, csu.owner, csu.repo, csu.sha)
+		if err != nil {
+			csu.logger.Error(err, "failed to get all commitStatuses for snapshot",
+				"snapshot.NameSpace", csu.snapshot.Namespace, "snapshot.Name", csu.snapshot.Name)
+			return nil, err
+		}
+		csu.allCommitStatusesCache = allCommitStatuses
+	}
+	return csu.allCommitStatusesCache, nil
+}
+
+// Authenticate Github Client with token secret ref defined in snapshot
+func (csu *CommitStatusUpdater) Authenticate(ctx context.Context, snapshot *applicationapiv1alpha1.Snapshot) error {
+	token, err := csu.getToken(ctx)
+	if err != nil {
+		csu.logger.Error(err, "failed to get token from snapshot",
+			"snapshot.NameSpace", snapshot.Namespace, "snapshot.Name", snapshot.Name)
+		return err
+	}
+
+	csu.ghClient.SetOAuthToken(ctx, token)
+	return nil
+}
+
+// createCommitStatusAdapterForSnapshot create a commitStatusAdapter used to create commitStatus on GitHub
+// https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#create-a-commit-status
+func (csu *CommitStatusUpdater) createCommitStatusAdapterForSnapshot(integrationTestStatusDetail intgteststat.IntegrationTestStatusDetail) (*github.CommitStatusAdapter, error) {
+	snapshot := csu.snapshot
+	scenarioName := integrationTestStatusDetail.ScenarioName
+	statusContext := NamePrefix + " / " + snapshot.Name + " / " + scenarioName
+
+	state, err := generateCommitState(integrationTestStatusDetail.Status)
+	if err != nil {
+		return nil, fmt.Errorf("unknown status %s for integrationTestScenario %s and snapshot %s/%s", integrationTestStatusDetail.Status, scenarioName, snapshot.Namespace, snapshot.Name)
+	}
+
+	description, err := generateSummary(integrationTestStatusDetail.Status, snapshot.Name, scenarioName)
+	if err != nil {
+		return nil, fmt.Errorf("unknown status %s for integrationTestScenario %s and snapshot %s/%s", integrationTestStatusDetail.Status, scenarioName, snapshot.Namespace, snapshot.Name)
+	}
```

Comment:
do we want to log the `err` variable too? goes for all the other similar return calls.

Reply:
The errors will be ultimately logged by the controller in any case, so I think it's good to leave as-is.

Reply:
Ack, thanks!